core/src/main/java/org/apache/iceberg/stats/StatsFormat.java

```diff
@@ -0,0 +1,143 @@
+import org.apache.iceberg.util.ByteBuffers;
+
+final class StatsFormat {
+  private StatsFormat() {
+  }
+
+  static final int CURRENT_FORMAT_VERSION = 1;
+
+  static final int MAGIC_AS_NUMBER_LE = new BigInteger(swapBytes(getMagic())).intValueExact();
+
+  static final int FLAG_COMPRESSED = 0b1;
+  static final int SUPPORTED_FLAGS = FLAG_COMPRESSED;
+
+  private static final StatsCompressionCodec FOOTER_COMPRESSION_CODEC = StatsCompressionCodec.LZ4;
+
+  static byte[] getMagic() {
+    return new byte[] {0x50, 0x46, 0x49, 0x53};
+  }
+
+  static void writeIntegerLittleEndian(OutputStream outputStream, int value) throws IOException {
+    outputStream.write(0xFF & value);
+    outputStream.write(0xFF & value >> 8);
+    outputStream.write(0xFF & value >> 16);
+    outputStream.write(0xFF & value >> 24);
+  }
+
+  static int readIntegerLittleEndian(byte[] data, int offset) {
+    return Byte.toUnsignedInt(data[offset]) |
+        (Byte.toUnsignedInt(data[offset + 1]) << 8) |
+        (Byte.toUnsignedInt(data[offset + 2]) << 16) |
+        (Byte.toUnsignedInt(data[offset + 3]) << 24);
+  }
+
+  static ByteBuffer compressFooterPayload(ByteBuffer payload) {
+    return compress(FOOTER_COMPRESSION_CODEC, payload);
+  }
+
+  static ByteBuffer decompressFooterPayload(ByteBuffer footer) {
+    return decompress(FOOTER_COMPRESSION_CODEC, footer);
+  }
+
+  static ByteBuffer compressBlob(StatsCompressionCodec codec, ByteBuffer data) {
+    return compress(codec, data);
+  }
+
+  static ByteBuffer decompressBlob(StatsCompressionCodec codec, ByteBuffer data) {
+    return decompress(codec, data);
+  }
+
+  private static ByteBuffer compress(StatsCompressionCodec codec, ByteBuffer input) {
+    Compressor compressor = getCompressor(codec);
+    ByteBuffer output = ByteBuffer.allocate(compressor.maxCompressedLength(input.remaining()));
+    compressor.compress(input, output);
+    output.flip();
+    return output;
+  }
+
+  private static Compressor getCompressor(StatsCompressionCodec codec) {
+    switch (codec) {
+      case LZ4:
+        // TODO currently not supported
+        break;
+      case ZSTD:
+        return new ZstdCompressor();
+    }
+    throw new UnsupportedOperationException("Unsupported codec: " + codec);
+  }
+
+  private static ByteBuffer decompress(StatsCompressionCodec codec, ByteBuffer input) {
+    switch (codec) {
+      case LZ4:
+        // TODO requires LZ4 frame decompressor, e.g. https://github.com/airlift/aircompressor/pull/142
+        break;
+
+      case ZSTD: {
+        byte[] inputBytes;
+        int inputOffset;
+        int inputLength;
+        if (input.hasArray()) {
+          inputBytes = input.array();
+          inputOffset = input.arrayOffset();
+          inputLength = input.remaining();
+        } else {
+          // TODO implement ZstdDecompressor.getDecompressedSize for ByteBuffer to avoid copying
+          inputBytes = ByteBuffers.toByteArray(input);
```

Comment:
Is there a maximum size that we could copy out in this case?

Reply:
The blob size. We didn't impose any limits on the size of blobs.

Reply:
This is where my other comment was supposed to go. This doesn't matter because the buffer isn't just used to get the size. It is also decompressed so it should be the whole blob.

Reply:
This is unclear to me, what should i do here?